### ISSUE TITLE
DOC: how to stream content inside my custom node? (not from chatOpenAI)

### ISSUE THREAD

**Issue by arlenkkk** [langchain-ai/langgraph#585]
### Issue with current documentation:

The langgrpah documentation includes two ways to stream.
1. graph.stream()
2. graph.stream_event()

The first way can only stream the output of a node.
The second way worked for ChatOpenAI method, but my situation is:
**I have a custom function, and i add it to langgraph node using "graph.add_nodes", the behavior of the function is to generate a long text like a LLM(but it is not a LLM), so I want to stream the output of my custom function, so that the user don't need to wait for the whole text to be generated.**

I read the docs for stream_event()， and find a similar approch is:
```
from langchain_core.runnables import RunnableLambda

async def reverse(s: str) -> str:
    return s[::-1]

chain = RunnableLambda(func=reverse)

events = [
    event async for event in chain.astream_events("hello", version="v1")
]
```
this will produce the following events (run_id has been omitted for brevity):
 ```
 [
                {
                    "data": {"input": "hello"},
                    "event": "on_chain_start",
                    "metadata": {},
                    "name": "reverse",
                    "tags": [],
                },
                {
                    "data": {"chunk": "olleh"},
                    "event": "on_chain_stream",
                    "metadata": {},
                    "name": "reverse",
                    "tags": [],
                },
                {
                    "data": {"output": "olleh"},
                    "event": "on_chain_end",
                    "metadata": {},
                    "name": "reverse",
                    "tags": [],
                },
            ]
```

But there is **only one chunk**, I want to generate **more chunks** like chunk1: o, chunk2: lle, chunks3: h, like the message chunks from a LLM, so that maybe i can stream the content inside my custom functions.

And I find the ToolNode from prebuilt maybe helpful in my situation, but the problem is there is also only one chunk from stream event.

### Idea or request for content:

Can you provide a doc for streaming any custom nodes? It will be more flexible for developers because not only LLM stream tokens but other functions may need stream tokens.

**Comment by hinthornw** [comment]
This is a great question! I added a section to the streaming tokens doc that should hopefully clarify how to stream output from an arbitrary custom function.

Tl;dr - it's very similar! While nodes themselves do not support token-level streaming (since the state updates have to occur between each superstep in the graph and we don't reaaally support returning generators as state updates), you can stream a generator as a `RunnableGenerator` (aka a runnable lambda) within a node, then those values will be included in `astream_events` as the `on_chain_stream` event kind.

**Comment by gbaian10** [comment]
I have similar problem.

<https://langchain-ai.github.io/langgraph/how-tos/streaming-tokens/#define-the-nodes>

In this example.

```python
async def call_model(state: State, config: RunnableConfig):
    messages = state["messages"]
    response = await model.ainvoke(messages, config)
    return {"messages": response}
```

Is it because of the use of `astream_events` that the underlying layer replaces `ainvoke` with a method similar to `astream`?

If you replace `ChatOpenAI` with `ChatAnthropic` in this example, you will get a warning and will not be able to stream output properly.

Is it because langgraph nests other runnables?  Otherwise using `ainvoke` seems a bit counterintuitive.

Please forgive me, I just learned langgraph and am not so familiar with contextvars.
